examples/daisyworld.jl

```diff
@@ -48,249 +60,269 @@ mutable struct Daisy <: AbstractAgent
-model = daisyworld(scenario = :ramp)
-
-# Then, let us initialize some dataframes for our model and agents. We are interested
-# in the global surface temperature, the current solar luminosity and populations of
-# each daisy breed. Notice that we made sure that `sum` has been given a default value
-# since this model is using `kill_agent!` (see [`run!`](@ref) for more details).
-
-global_temperature(model) = mean(model.temperature)
-mdata = [global_temperature, :solar_luminosity]
-model_df = init_model_dataframe(model, mdata)
-
-white(agent) = agent.breed == :white
-black(agent) = agent.breed == :black
-total(v) = length(v) == 0 ? 0.0 : sum(v)
-adata = [(white, total), (black, total)]
-agent_df = init_agent_dataframe(model, adata)
-nothing # hide
-
-# Now we can evolve our model and observe what happens
-
-anim = @animate for t in 1:900
-    step!(model, agent_step!, model_step!, 1)
-    collect_model_data!(model_df, model, mdata, t)
-    collect_agent_data!(agent_df, model, adata, t)
-    heatmap(
-        1:model.space.dimensions[1],
-        1:model.space.dimensions[2],
-        transpose(reshape(model.temperature, model.space.dimensions));
-        clims = (-50, 110),
-        colorbar_title = "Temperature",
-    )
-    scatter!(
-        [a.pos for a in allagents(model)];
-        marker = (:circle, 5),
-        markercolor = [a.breed for a in allagents(model)],
-        label = :none,
-        showaxis = false,
-    )
-end
-gif(anim, "daisyworld.gif", fps = 10)
-
-# Very interesting! But why is this all happening? Luckily we have collected some useful
-# data, so now if we plot our different properties over the same time period, we can see
-# how each of the values effect Daisyworld as a whole.
-
-p1 = plot(model_df[!, :solar_luminosity], legend = false, ylabel = "Solar Luminosity")
-p2 = plot(model_df[!, :global_temperature], legend = false, ylabel = "Global Temperature")
-p3 = plot(
-    [agent_df[!, aggname(white, total)], agent_df[!, aggname(black, total)]],
-    legend = false,
-    ylabel = "Population",
+model = daisyworld()
+
+# To visualize we need to define the necessary functions for [`plotabm`](@ref).
+# The daisies will obviously be black or white, but the land will have a color
+# that reflects its temperature, with -50 darkest and 100 ᵒC brightest color
+
+daisycolor(a::Daisy) = a.breed
+const landcolor = cgrad(:thermal)
+daisycolor(a::Land) = landcolor[(a.temperature+50)/150]
+
+# And we plot daisies as circles, and land patches as squares
+daisyshape(a::Daisy) = :circle
+daisysize(a::Daisy) = 8
+daisyshape(a::Land) = :square
+daisysize(a::Land) = 13.5
+
+# Notice that we want to ensure that the `Land` patches are always plotted first.
+plotsched = by_type((Land, Daisy), false)
+
+plotkwargs = (
+    ac = daisycolor, am = daisyshape, as = daisysize,
+    scheduler = plotsched, aspect_ratio = 1, size = (600, 600), showaxis = false,
 )
-plot(p1, p2, p3, layout = (3, 1), size = (500, 800))
+
+p = plotabm(model; plotkwargs...)
```

Comment:
The land size is not correct, both here and in the next two plots

![landissue](https://user-images.githubusercontent.com/3059508/88203618-63122080-cc4a-11ea-9bde-f309585f5de6.png)

Reply:
I have a suspicion that it may depend on screen resolution.... But I'll test again.

Reply:
Turns out it strongly depends on the backend (I use pyplot)

Reply:
Ah, fair enough. We have pyplot available - some of the other examples are built with it, but this guy takes forever on Travis with it enabled.